chainer/functions/cross_covariance.py

```diff
@@ -0,0 +1,109 @@
+import numpy
+
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+class CrossCovariance(function.Function):
+
+    """Cross-covariance loss."""
+
+    def __init__(self, use_cudnn=True):
+        self.use_cudnn = use_cudnn
+        self.y_centered = None
+        self.z_centered = None
+        self.covariance = None
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 2)
+        z_type, y_type = in_types
+
+        type_check.expect(
+            z_type.dtype == numpy.float32,
+            z_type.ndim == 2,
+            y_type.dtype == numpy.float32,
+            y_type.ndim == 2,
+
+            z_type.shape[0] == y_type.shape[0]
+        )
+
+    def check_type_backward(self, in_types, out_types):
+        type_check.expect(
+            in_types.size() == 2,
+            out_types.size() == 1,
+        )
+        y_in_type, z_in_type = in_types
+        out_type, = out_types
+
```

Comment:
To be precise, `out_type.ndim` is should be 0. I leave it to you if this fix will be applied because we are thinking `check_type_backward` should be removed from `Function` class.  See #209 for details.